druid-shell/src/platform/gtk/window.rs

```diff
@@ -982,7 +990,30 @@ impl WindowHandle {
     }
 
     pub fn handle_titlebar(&self, _val: bool) {
-        warn!("WindowHandle::handle_titlebar is currently unimplemented for gtk.");
+        warn!("WindowHandle::handle_titlebar is currently unimplemented for gtk. Use WindowHandle::begin_move_drag instead.");
+    }
+
+    /// Tell the window manager to start a window drag using current mouse position.
+    pub fn begin_move_drag(&self) {
+        if let Some(state) = self.state.upgrade() {
+            match state.last_mouse_button_press_event.take() {
+                //Consume the mouse button event.
+                Some(event) if event.get_event_type() == gdk::EventType::ButtonPress => {
+                    state.window.begin_move_drag(
+                        event.get_button() as i32,
+                        event.get_root().0 as i32,
+                        event.get_root().1 as i32,
+                        event.get_time(),
+                    )
+                }
+                _ => warn!("Cannot begin window drag if no mouse button is currently down."),
+            }
+        }
+    }
+
+    /// Tell the window manager to end the current window drag, if any.
+    pub fn end_move_drag(&self) {
+        //Nothing to do on gtk. Window drag immediately ends when mouse button is released.
```

Comment:
Is it ever valid to end the drag earlier? If not, why not just delete this? It does nothing on other platforms anyway...

Reply:
I wanted to make sure it had a reverse operation (like `open` and `close`, `start` and `stop`, `begin` and `end`), but in retrospect, that was a bad idea.